standard_sequence/classy_sequence.py

```diff
@@ -861,263 +318,528 @@ def _do_molasses_tof_sequence(self, t, reset_mot = False):
+            self.D2Lasers_obj.ramp_repump_freq(
+                t, D2Lasers.CONST_TA_VCO_RAMP_TIME, shot_globals.op_repump_pumping_detuning
+            )
             # Do a sigma+ pulse
             # TODO: is shot_globals.op_ramp_delay just extra fudge time? can it be eliminated?
-            t += max(CONST_TA_VCO_RAMP_TIME, shot_globals.op_ramp_delay)
-            t = self.D2Lasers_obj.do_pulse(t - CONST_SHUTTER_TURN_ON_TIME,
-                                       shot_globals.op_repump_time,
-                                       ShutterConfig.OPTICAL_PUMPING_FULL,
-                                       shot_globals.op_ta_power,
-                                       shot_globals.op_repump_power,)
-            # Need to turn off the TA before repump, Sam claims this timing should work
-            assert shot_globals.op_ta_time < shot_globals.op_repump_time, "TA time should be shorter than repump for pumping to F=4"
-            # TODO: test this timing
-            self.D2Lasers_obj.ta_aom_off(t + shot_globals.op_ta_time - shot_globals.op_repump_time)
-            # Close the shutters
-            self.D2Lasers_obj.update_shutters(t, ShutterConfig.NONE)
-            t += CONST_SHUTTER_TURN_OFF_TIME
+            t += max(D2Lasers.CONST_TA_VCO_RAMP_TIME, shot_globals.op_ramp_delay)
+            t, t_aom_start = self.D2Lasers_obj.do_pulse(
+                t,
+                shot_globals.op_repump_time,
+                ShutterConfig.OPTICAL_PUMPING_FULL,
+                shot_globals.op_ta_power,
+                shot_globals.op_repump_power,
+                close_all_shutters=close_all_shutters
+            )
 
-            return t
+            t_aom_off = t_aom_start + shot_globals.op_repump_time
 
+            assert (
+                shot_globals.op_ta_time < shot_globals.op_repump_time
+            ), "TA time should be shorter than repump for pumping to F=4"
+            # TODO: test this timing
+            self.D2Lasers_obj.ta_aom_off(
+                t_aom_start + shot_globals.op_ta_time
+            )
+            # Close the shutters
+            return t, t_aom_off
         else:
             raise NotImplementedError("This optical pumping method is not implemented")
 
-    def depump_to_F3(self, t):
+
+
+    def depump_ta_pulse(self, t, close_all_shutters = True):
+        # This method is only depump pulse with ta alone to pump atom from F=4 -> F=3
+        # This will determine the minimum time offset between repump and ta when doing depump_to_F3
+        # This will also be used in dark state measurement
+        if self.BField_obj.mot_coils_on:
+            _ = self.BField_obj.switch_mot_coils(t)
+
+        t = self.D2Lasers_obj.ramp_ta_freq(t, D2Lasers.CONST_TA_VCO_RAMP_TIME, CONST_TA_PUMPING_DETUNING)
+        #t += max(D2Lasers.CONST_TA_VCO_RAMP_TIME, shot_globals.op_ramp_delay)
+        t, _ = self.D2Lasers_obj.do_pulse(
```

Comment:
Add description of why OpticalPumping.Full shutter config is needed, when this is only a TA a pulse. I couldn't figure out why this function is unique compared to the depump to F3 function below.

Reply:
# The shutter configuration needs to be optical_pumping_full
        # to make sure no shutter switch from the depump_to_F3/pump_to_F4
        # sequence, this allow the two pulse sequence purely switched
        # with aom so that they are next to each other